bin/juttle-engine-client

```diff
@@ -109,11 +109,41 @@ function watch_path(opts) {
     }
 }
 
+// Given a complete path, which reflects a path on the local
+// filesystem, figure out the right path suffix to use relative to the
+// juttle-service's root path. Returns a promise that resolves with
+// the correct path suffix on or rejects if the local path and the
+// service's root path don't share a prefix.
+function reconcile_path(juttle_engine, local_path) {
+    let idx;
+
+    local_path = path.normalize(local_path);
+
+    return rp(`http://${juttle_engine}/api/v0/config/`)
+    .then(function(body) {
+        return JSON.parse(body);
+    })
+    .then(function(config) {
+        if (!_.has(config, 'root')) {
+            throw new Error('Server has no configured root directory');
+        }
+        // The root directory must be a prefix of the local path.
+        idx = local_path.indexOf(config.root);
+        if (idx !== 0) {
+            throw new Error(`Local path (${local_path}) is not a path below server root path (${config.root})`);
+        } else {
+            let relative_to_root = local_path.substring(config.root.length);
+            return relative_to_root;
+        }
+    });
+}
```

Comment:
do we want this to be possible for juttle-service to not have a root_path? If this is the case then juttle-service/path should always throw an error. I've put up a pr that would not let you initialize a juttle-service without a root_directory https://github.com/juttle/juttle-service/pull/68